diff --git a/frameworks/base/core/java/android/bluetooth/BluetoothSocket.java b/frameworks/base/core/java/android/bluetooth/BluetoothSocket.java
index d10eaea..61f660c 100644
--- a/frameworks/base/core/java/android/bluetooth/BluetoothSocket.java
+++ b/frameworks/base/core/java/android/bluetooth/BluetoothSocket.java
@@ -461,8 +461,14 @@ public final class BluetoothSocket implements Closeable {
                     mSocket.close();
                     mSocket = null;
                 }
-                if(mPfd != null)
-                    mPfd.detachFd();
+
+                if(mPfd != null) {
+                    try {
+                        mPfd.detachFd();
+                    } catch (IllegalArgumentException e) {
+                        Log.w(TAG, "ignore IllegalArgumentException");
+                    }
+                }
            }
         }
     }
diff --git a/frameworks/base/core/java/android/net/LocalSocketImpl.java b/frameworks/base/core/java/android/net/LocalSocketImpl.java
index b2ee50a..d968cf1 100644
--- a/frameworks/base/core/java/android/net/LocalSocketImpl.java
+++ b/frameworks/base/core/java/android/net/LocalSocketImpl.java
@@ -39,8 +39,6 @@ class LocalSocketImpl
 
     /** null if closed or not yet created */
     private FileDescriptor fd;
-    /** whether fd is created internally */
-    private boolean mFdCreatedInternally;
 
     // These fields are accessed by native code;
     /** file descriptor array received during a previous read */
@@ -249,7 +247,6 @@ class LocalSocketImpl
             }
             try {
                 fd = Libcore.os.socket(OsConstants.AF_UNIX, osType, 0);
-                mFdCreatedInternally = true;
             } catch (ErrnoException e) {
                 e.rethrowAsIOException();
             }
@@ -263,7 +260,7 @@ class LocalSocketImpl
      */
     public void close() throws IOException {
         synchronized (LocalSocketImpl.this) {
-            if ((fd == null) || (mFdCreatedInternally == false)) {
+            if ((fd == null) ) {
                 fd = null;
                 return;
             }
diff --git a/packages/apps/Bluetooth/src/com/android/bluetooth/btservice/AdapterService.java b/packages/apps/Bluetooth/src/com/android/bluetooth/btservice/AdapterService.java
index b1d8875..6403afd 100755
--- a/packages/apps/Bluetooth/src/com/android/bluetooth/btservice/AdapterService.java
+++ b/packages/apps/Bluetooth/src/com/android/bluetooth/btservice/AdapterService.java
@@ -290,6 +290,11 @@ public class AdapterService extends Service {
         mAdapterProperties.init(mRemoteDevices);
 
         if (DBG) {debugLog("processStart(): Make Bond State Machine");}
+        if(mBondStateMachine != null) //fix leak
+        {
+            mBondStateMachine.doQuit();
+            mBondStateMachine.cleanup();
+        }
         mBondStateMachine = BondStateMachine.make(this, mAdapterProperties, mRemoteDevices);
 
         mJniCallbacks.init(mBondStateMachine,mRemoteDevices);
