From ee000f84d6308b51d68cacdb2ece0ae7e56973ae Mon Sep 17 00:00:00 2001
From: ranhui_xia <ranhui_xia@realsil.com.cn>
Date: Thu, 12 Dec 2013 11:43:28 +0800
Subject: [PATCH 08/13] Realtek BT: Save device name change information

Signed-off-by: ranhui_xia <ranhui_xia@realsil.com.cn>
---
 btif/src/btif_core.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/btif/src/btif_core.c b/btif/src/btif_core.c
index ccaef17..fa607a7 100644
--- a/btif/src/btif_core.c
+++ b/btif/src/btif_core.c
@@ -1206,6 +1206,11 @@ bt_status_t btif_set_adapter_property(const bt_property_t *property)
                 BTA_DmSetDeviceName((char *)bd_name);
 
                 storage_req_id = BTIF_CORE_STORAGE_ADAPTER_WRITE;
+                //REALTEK ADD START
+                btif_config_set_str("Local", "Adapter", "Name", (char *)bd_name);
+                btif_config_save();
+                //REALTEK ADD END
+
             }
             break;
 
-- 
1.8.5.rc2

