diff --git a/services/core/java/com/android/server/BluetoothManagerService.java b/services/core/java/com/android/server/BluetoothManagerService.java
old mode 100644
new mode 100755
index 50bd544..73702d5
--- a/services/core/java/com/android/server/BluetoothManagerService.java
+++ b/services/core/java/com/android/server/BluetoothManagerService.java
@@ -1680,6 +1680,9 @@ class BluetoothManagerService extends IBluetoothManager.Stub {
                                                         PackageManager.FEATURE_BLUETOOTH_LE)) {
                             Intent i = new Intent(IBluetoothGatt.class.getName());
                             doBind(i, mConnection, Context.BIND_AUTO_CREATE | Context.BIND_IMPORTANT, UserHandle.CURRENT);
+                        } else {
+                            Log.d(TAG, "Calling BluetoothGattServiceUp");
+                            onBluetoothGattServiceUp();
                         }
                     }
                     sendBleStateChanged(prevState, newState);
