From 15d8789109493495e9e5b4a2c9d9029ea9584157 Mon Sep 17 00:00:00 2001
From: wangchihhs <wangchihhs@wangchihhs-OptiPlex-755.(none)>
Date: Fri, 17 Jan 2014 14:45:23 +0800
Subject: [PATCH] fix opp resend file

---
 .../android/bluetooth/opp/BluetoothOppUtility.java |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/com/android/bluetooth/opp/BluetoothOppUtility.java b/src/com/android/bluetooth/opp/BluetoothOppUtility.java
index 7e03281..5ff2546 100755
--- a/src/com/android/bluetooth/opp/BluetoothOppUtility.java
+++ b/src/com/android/bluetooth/opp/BluetoothOppUtility.java
@@ -302,6 +302,9 @@ public class BluetoothOppUtility {
         values.put(BluetoothShare.MIMETYPE, transInfo.mFileType);
         values.put(BluetoothShare.DESTINATION, transInfo.mDestAddr);
 
+        Uri uri = Uri.parse(transInfo.mFileUri);
+        putSendFileInfo(uri, BluetoothOppSendFileInfo.generateFileInfo(context, uri, transInfo.mFileType));
+
         final Uri contentUri = context.getContentResolver().insert(BluetoothShare.CONTENT_URI,
                 values);
         if (V) Log.v(TAG, "Insert contentUri: " + contentUri + "  to device: " +
-- 
1.7.9.5

