From d6db43e5e77ad62d19fcb873ea387373511ecacf Mon Sep 17 00:00:00 2001
From: wangchihhs <wangchihhs@wangchihhs-OptiPlex-755.(none)>
Date: Fri, 17 Jan 2014 14:47:44 +0800
Subject: [PATCH] add delay after opp transfer

---
 .../opp/BluetoothOppObexServerSession.java         |   13 ++++++++--
 1 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/src/com/android/bluetooth/opp/BluetoothOppObexServerSession.java b/src/com/android/bluetooth/opp/BluetoothOppObexServerSession.java
index 3512fad..55cf6cd 100755
--- a/src/com/android/bluetooth/opp/BluetoothOppObexServerSession.java
+++ b/src/com/android/bluetooth/opp/BluetoothOppObexServerSession.java
@@ -570,8 +570,7 @@ public class BluetoothOppObexServerSession extends ServerRequestHandler implemen
 
     @Override
     public void onClose() {
-        if (V) Log.v(TAG, "release WakeLock");
-        releaseWakeLocks();
+        PowerManager pm = (PowerManager) mContext.getSystemService(Context.POWER_SERVICE);
 
         /* onClose could happen even before start() where mCallback is set */
         if (mCallback != null) {
@@ -580,5 +579,15 @@ public class BluetoothOppObexServerSession extends ServerRequestHandler implemen
             msg.obj = mInfo;
             msg.sendToTarget();
         }
+
+        boolean isScreenOn = pm.isScreenOn();
+
+        if (!isScreenOn ) {
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+            }
+        }
+        releaseWakeLocks();
     }
 }
-- 
1.7.9.5

