From 874dc0e323f7f65e4cc85e5b7a421cc5c466a10d Mon Sep 17 00:00:00 2001
From: wangchihhs <wangchihhs@wangchihhs-OptiPlex-755.(none)>
Date: Fri, 17 Jan 2014 15:01:03 +0800
Subject: [PATCH] add pair retry

---
 btif/src/btif_dm.c                |   40 +++++++++++++++++++++++++++++-----------
 1 files changed, 29 insertions(+), 11 deletions(-)

diff --git a/btif/src/btif_dm.c b/btif/src/btif_dm.c
index 5517f55..9855a03 100755
--- a/btif/src/btif_dm.c
+++ b/btif/src/btif_dm.c
@@ -63,6 +63,7 @@
 #define BTIF_DM_DEFAULT_INQ_MAX_RESULTS     0
 #define BTIF_DM_DEFAULT_INQ_MAX_DURATION    10
 #define BTIF_DM_MAX_SDP_ATTEMPTS_AFTER_PAIRING 2
+#define BOND_RETRY_MAXIMUM_RETRY_TIMES      3 //add by realtek, For retry device bond
 
 #define PROPERTY_PRODUCT_MODEL "ro.product.model"
 #define DEFAULT_LOCAL_NAME_MAX  31
@@ -122,6 +123,10 @@ typedef struct
     BT_OCTET16 sp_r;
     BD_ADDR  oob_bdaddr;  /* peer bdaddr*/
 } btif_dm_oob_cb_t;
+//add by realtek start, For retry device bond
+bt_bdaddr_t tempAddr;
+int bond_retry_times = 0;
+//add by realtek end, For retry device bond
 #define BTA_SERVICE_ID_TO_SERVICE_MASK(id)       (1 << (id))
 
 /* This flag will be true if HCI_Inquiry is in progress */
@@ -341,17 +346,28 @@ static void bond_state_changed(bt_status_t status, bt_bdaddr_t *bd_addr, bt_bond
     }
     BTIF_TRACE_DEBUG3("%s: state=%d prev_state=%d", __FUNCTION__, state, pairing_cb.state);
 
-    HAL_CBACK(bt_hal_cbacks, bond_state_changed_cb, status, bd_addr, state);
-
-    if (state == BT_BOND_STATE_BONDING)
-    {
-        pairing_cb.state = state;
-        bdcpy(pairing_cb.bd_addr, bd_addr->address);
-    }
-    else
-    {
-        memset(&pairing_cb, 0, sizeof(pairing_cb));
-    }
+    //add by realtek start, Add bonding retry mechanism
+    if (pairing_cb.state == BT_BOND_STATE_BONDING && 
+        state == BT_BOND_STATE_NONE &&
+        status == BT_STATUS_RMT_DEV_DOWN &&
+        bond_retry_times < BOND_RETRY_MAXIMUM_RETRY_TIMES) {
+		    bond_retry_times++;
+		    btif_dm_cb_create_bond(&tempAddr);
+	  //add by realtek end, Add bonding retry mechanism
+	  } else {
+		    bond_retry_times = 0;
+		    HAL_CBACK(bt_hal_cbacks, bond_state_changed_cb, status, bd_addr, state);
+
+		    if (state == BT_BOND_STATE_BONDING)
+		    {
+			      pairing_cb.state = state;
+			      bdcpy(pairing_cb.bd_addr, bd_addr->address);
+		    }
+		    else
+		    {
+			      memset(&pairing_cb, 0, sizeof(pairing_cb));
+		    }
+	  }
 
 }
 
@@ -1853,6 +1870,8 @@ bt_status_t btif_dm_create_bond(const bt_bdaddr_t *bd_addr)
 {
     bdstr_t bdstr;
 
+    memcpy(&tempAddr, bd_addr, sizeof(bt_bdaddr_t));
+
     BTIF_TRACE_EVENT2("%s: bd_addr=%s", __FUNCTION__, bd2str((bt_bdaddr_t *) bd_addr, &bdstr));
     if (pairing_cb.state != BT_BOND_STATE_NONE)
         return BT_STATUS_BUSY;

1.7.9.5

