From 7ac5167a8aa4be8a0d180336985dd1018939af18 Mon Sep 17 00:00:00 2001
From: wangchihhs <wangchihhs@wangchihhs-OptiPlex-755.(none)>
Date: Fri, 17 Jan 2014 14:58:11 +0800
Subject: [PATCH] fix hid could not pair

---
 btif/src/btif_storage.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/btif/src/btif_storage.c b/btif/src/btif_storage.c
index 2190b1c..2ee0fbb 100755
--- a/btif/src/btif_storage.c
+++ b/btif/src/btif_storage.c
@@ -908,7 +908,10 @@ bt_status_t btif_storage_load_bonded_hid_info(void)
         kpos = btif_config_next_key(kpos, "Remote", kname, &kname_size);
         BTIF_TRACE_DEBUG2("Remote device:%s, size:%d", kname, kname_size);
         int value;
-        if(btif_config_get_int("Remote", kname, "HidAttrMask", &value))
+        int linkkey_type;
+        if(btif_config_get_int("Remote", kname, "HidAttrMask", &value) &&
+           (btif_config_get_int("Remote", kname, "LinkKeyType", &linkkey_type) ||
+            btif_config_exist("Remote", kname, "LE_KEY_PENC")))
         {
             attr_mask = (uint16_t)value;
 
-- 
1.7.9.5

