diff --git a/services/java/com/android/server/BluetoothManagerService.java b/services/java/com/android/server/BluetoothManagerService.java
old mode 100644
new mode 100755
index 546324a..90d67f7
--- a/services/java/com/android/server/BluetoothManagerService.java
+++ b/services/java/com/android/server/BluetoothManagerService.java
@@ -44,6 +44,13 @@ import android.os.SystemClock;
 import android.os.UserHandle;
 import android.provider.Settings;
 import android.util.Log;
+//add for hot plug by jim
+import android.hardware.usb.UsbDevice;
+import android.hardware.usb.UsbInterface;
+import android.hardware.usb.UsbManager;
+import android.os.DeadObjectException;
+//end by jim
+
 class BluetoothManagerService extends IBluetoothManager.Stub {
     private static final String TAG = "BluetoothManagerService";
     private static final boolean DBG = true;
@@ -122,6 +129,10 @@ class BluetoothManagerService extends IBluetoothManager.Stub {
     private final BluetoothHandler mHandler;
     private int mErrorRecoveryRetryCounter;
 
+    //add for hot plug by jim
+    private UsbDevice mDevice;
+    //end by jim
+
     private void registerForAirplaneMode(IntentFilter filter) {
         final ContentResolver resolver = mContext.getContentResolver();
         final String airplaneModeRadios = Settings.Global.getString(resolver,
@@ -187,7 +198,40 @@ class BluetoothManagerService extends IBluetoothManager.Stub {
                     if (DBG) Log.d(TAG,"Retrieving Bluetooth Adapter name and address...");
                     getNameAndAddress();
                 }
-            }
+            }/*add for hot plug by jim*/else if(UsbManager.ACTION_USB_DEVICE_DETACHED.equals(action)) {
+                Log.d(TAG, "usb device detach detected");
+                UsbDevice device = (UsbDevice)intent.getParcelableExtra(UsbManager.EXTRA_DEVICE);
+                if (device != null && device.equals(mDevice)) {
+                    Log.e(TAG,"RTK USB BT device detached.");
+                    mDevice = null;
+                    disable(true);
+                }
+            }else if(UsbManager.ACTION_USB_DEVICE_ATTACHED.equals(action)) {
+                Log.d(TAG, "new usb device attach detected");
+                UsbDevice device  = (UsbDevice)intent.getParcelableExtra(UsbManager.EXTRA_DEVICE);
+                if((device != null) && (device.getVendorId() == 0x0BDA)) {
+                    int count = device.getInterfaceCount();
+                    if(count > 0) {
+                        for(int i = 0; i < count; i++) {
+                            UsbInterface usbIf = device.getInterface(i);
+                            if(usbIf != null) {
+                                int devClass = usbIf.getInterfaceClass();
+                                int devSubClass = usbIf.getInterfaceSubclass();
+                                int devProtocol = usbIf.getInterfaceProtocol();
+
+                                if((devClass == 0xE0) && (devSubClass == 0x01) && (devProtocol == 0x01)) {
+                                    Log.d(TAG, "RTK USB BT device attached.");
+                                    mDevice = device;
+                                    enable();
+                                    Message msg = mHandler.obtainMessage(MESSAGE_BLUETOOTH_STATE_CHANGE,BluetoothAdapter.STATE_OFF,BluetoothAdapter.STATE_TURNING_ON);
+                                    mHandler.sendMessage(msg);
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+            }/*end by jim*/
         }
     };
 
@@ -211,6 +255,10 @@ class BluetoothManagerService extends IBluetoothManager.Stub {
         IntentFilter filter = new IntentFilter(Intent.ACTION_BOOT_COMPLETED);
         filter.addAction(BluetoothAdapter.ACTION_LOCAL_NAME_CHANGED);
         filter.addAction(Intent.ACTION_USER_SWITCHED);
+        //add for hot plug by jim
+        filter.addAction(UsbManager.ACTION_USB_DEVICE_ATTACHED);
+        filter.addAction(UsbManager.ACTION_USB_DEVICE_DETACHED);
+        //end by jim
         registerForAirplaneMode(filter);
         mContext.registerReceiver(mReceiver, filter);
         loadStoredNameAndAddress();
