From 23dcf203a291e0a49dec2793b32f7f8c5dd2fa24 Mon Sep 17 00:00:00 2001
From: champion_chen <champion_chen@realsil.com.cn>
Date: Wed, 16 Apr 2014 17:43:31 +0800
Subject: [PATCH] Realtek BT: Set default scan mode standard

Signed-off-by: champion_chen <champion_chen@realsil.com.cn>
---
 include/bt_target.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/bt_target.h b/include/bt_target.h
index 8dea76a..2af663b 100644
--- a/include/bt_target.h
+++ b/include/bt_target.h
@@ -947,7 +947,9 @@ and USER_HW_DISABLE_API macros */
 
 /* The default scan mode */
 #ifndef BTM_DEFAULT_SCAN_TYPE
-#define BTM_DEFAULT_SCAN_TYPE       BTM_SCAN_TYPE_INTERLACED
+//#define BTM_DEFAULT_SCAN_TYPE       BTM_SCAN_TYPE_INTERLACED
+#define BTM_DEFAULT_SCAN_TYPE       BTM_SCAN_TYPE_STANDARD //RTK Modified
+
 #endif
 
 /* Should connections to unknown devices be allowed when not discoverable? */
-- 
1.8.5.rc2

