diff --git a/stack/btm/btm_ble_gap.c b/stack/btm/btm_ble_gap.c
index ccea5e3..05b7d3a 100755
--- a/stack/btm/btm_ble_gap.c
+++ b/stack/btm/btm_ble_gap.c
@@ -1765,6 +1765,11 @@ static void btm_ble_process_adv_pkt_cont(BD_ADDR bda, UINT8 addr_type, UINT8 evt
 #endif
 
     p_i = btm_inq_db_find (bda);
+    if (!btm_ble_is_discoverable(bda, evt_type, p))
+    {
+        BTM_TRACE_ERROR0("rtk discard adv pkt");
+        return;
+    }
 
     /* Check if this address has already been processed for this inquiry */
     if (btm_inq_find_bdaddr(bda))
